src/cascadia/TerminalSettingsModel/ActionMap.cpp

```diff
@@ -277,88 +302,101 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
+        //   { "command": "copy", "keys": "ctrl+shift+c" } --> update oldCmd
+        const auto actionID{ Hash(cmd.ActionAndArgs()) };
         const auto& actionPair{ _ActionMap.find(actionID) };
-        if (actionPair != _ActionMap.end())
-        {
-            // We're adding an action that already exists in our layer.
-            // Record it so that we update it with any new information.
-            oldCmd = actionPair->second;
-        }
-        else
+        if (actionPair == _ActionMap.end())
         {
             // add this action in for the first time
             _ActionMap.emplace(actionID, cmd);
         }
-
-        // Now check if this action was introduced in another layer.
-        const auto& consolidatedActionPair{ _ConsolidatedActions.find(actionID) };
-        if (consolidatedActionPair != _ConsolidatedActions.end())
+        else
         {
-            // We're adding an action that already existed on a different layer.
+            // We're adding an action that already exists in our layer.
             // Record it so that we update it with any new information.
-            consolidatedCmd = consolidatedActionPair->second;
+            oldCmd = actionPair->second;
         }
-        else
+
+        // Masked Actions
+        //
+        // Example:
+        //   parent:  { "command": "copy", "keys": "ctrl+c" }       --> add the action to parent._ActionMap
+        //   current: { "command": "copy", "keys": "ctrl+shift+c" } --> look through parents for the "ctrl+c" binding, add it to _MaskedActions
+        //            { "command": "copy", "keys": "ctrl+ins" }     --> this should already be in _MaskedActions
+
+        // Now check if this action was introduced in another layer.
+        const auto& maskedActionPair{ _MaskedActions.find(actionID) };
+        if (maskedActionPair == _MaskedActions.end())
         {
-            // Check if we need to add this to our list of consolidated commands.
+            // Check if we need to add this to our list of masked commands.
             FAIL_FAST_IF(_parents.size() > 1);
             for (const auto& parent : _parents)
             {
                 // NOTE: This only checks the layer above us, but that's ok.
-                //       If we had to find one from a layer above that, parent->_ConsolidatedActions
+                //       If we had to find one from a layer above that, parent->_MaskedActions
                 //       would have found it, so we inherit it for free!
                 const auto& inheritedCmd{ parent->_GetActionByID(actionID) };
                 if (inheritedCmd.has_value() && inheritedCmd.value())
                 {
                     const auto& inheritedCmdImpl{ get_self<Command>(inheritedCmd.value()) };
-                    consolidatedCmd = *inheritedCmdImpl->Copy();
-                    _ConsolidatedActions.emplace(actionID, consolidatedCmd);
+                    maskedCmd = *inheritedCmdImpl->Copy();

```

Comment:
riffing on the naming thing I mentioned in Teams-
this action *masks* the one below it. It is a `masking` action :smile:

The `masked` one is `inheritedCmd` (!)